vpr/src/place/timing_place_lookup.cpp

```diff
@@ -532,63 +532,48 @@ float delay_reduce(std::vector<float>& delays, e_reducer reducer) {
 
 /* Take the average of the valid neighboring values in the matrix.
  * use interpolation to get the right answer for empty blocks*/
-static float find_neightboring_average(vtr::Matrix<float>& matrix, int x, int y) {
+static float find_neightboring_average(vtr::Matrix<float>& matrix, int x, int y, int max_distance) {
     float sum = 0;
     int counter = 0;
     int endx = matrix.end_index(0);
     int endy = matrix.end_index(1);
 
     int delx, dely;
 
-    for (delx = x - 1; delx <= x + 1; delx++) {
-        for (dely = y - 1; dely <= y + 1; dely++) {
-            //check out of bounds
-            if ((delx == x - 1 && dely == y + 1) || (delx == x + 1 && dely == y + 1) || (delx == x - 1 && dely == y - 1) || (delx == x + 1 && dely == y - 1)) {
-                continue;
-            }
-            if (delx < 0 || dely < 0 || delx >= endx || dely >= endy || (delx == x && dely == y)) {
-                continue;
-            }
-            if (matrix[delx][dely] == EMPTY_DELTA || matrix[delx][dely] == IMPOSSIBLE_DELTA) {
-                continue;
-            }
-            counter++;
-            sum += matrix[delx][dely];
-        }
-    }
+    for (int distance = 1; distance <= max_distance; ++distance) {
+        for (delx = x - distance; delx <= x + distance; delx++) {
+            for (dely = y - distance; dely <= y + distance; dely++) {
+                // Check distance constraint
+                if (abs(delx - x) + abs(dely - y) > distance) {
+                    continue;
+                }
 
-    if (counter == 0) {
-        //take in more values for more accuracy
-        sum = 0;
-        counter = 0;
-        for (delx = x - 1; delx <= x + 1; delx++) {
-            for (dely = y - 1; dely <= y + 1; dely++) {
                 //check out of bounds
-                if ((delx == x && dely == y) || delx < 0 || dely < 0 || delx >= endx || dely >= endy) {
+                if (delx < 0 || dely < 0 || delx >= endx || dely >= endy || (delx == x && dely == y)) {
                     continue;
                 }
+
                 if (matrix[delx][dely] == EMPTY_DELTA || matrix[delx][dely] == IMPOSSIBLE_DELTA) {
                     continue;
                 }
                 counter++;
                 sum += matrix[delx][dely];
             }
         }
+        if (counter != 0) {
+            return sum / (float)counter;
+        }
     }
 
-    if (counter != 0) {
-        return sum / (float)counter;
-    } else {
-        return 0;
-    }
+    return IMPOSSIBLE_DELTA;
 }
 
 static void fix_empty_coordinates(vtr::Matrix<float>& delta_delays) {
     //Set any empty delta's to the average of it's neighbours
```

Comment:
Add why to the comment:
- Why do these empty deltas occur, and why do we need to fill them?  
- I think the reason is that our sampling of the routing may miss some options, as we only sample a subset of the routing and assume translational invariance. But leaving empty deltas (-1 values) in the delay estimation matrix can lead the placement algorithm to do strange things later, so filling these in with the average around them is much better.
- Should explain why 2 is chosen as the max_distance (probably heuristic, but explain the intuition & testing)